pkg/kotsadm/objects/scripts/copy-minio-client.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -e
+
+# This script copies the minio client binary to a dedicated migration directory
+# This is necessary because the minio client binary is not available in the minio container
+
+# check if the migration has already been completed
+if [ -f /export/.migration ];
+then
+    MIGRATION_DATE=$(cat /export/.migration)
+    echo "migration already completed at $MIGRATION_DATE, no-op"
+    exit 0
+fi
+
+# validate environment variables
+if [ -z $KOTSADM_MINIO_MIGRATION_DIR ]; then
+    echo 'KOTSADM_MINIO_MIGRATION_DIR not set'
+    exit 1
+fi
+
+# clean the migration directory
+shopt -s dotglob
+rm -rfv $KOTSADM_MINIO_MIGRATION_DIR/*
```

Comment:
hmm, I know i suggested cleaning up this directory earlier, but consider this scenario:

during the migration, the old data is copied out of the pvc successfully, then, copying the new data into the pvc fails. at that point, there is a copy of the old data in the migration emptydir volume, and since kubelet will only restart the container and not the pod, i _believe_ the emptydir will still contain that data after containers restarts, which is valuable IMO and could be utilized to prevent data loss in some cases. what happens if `mc mirror` is called again and the pvc does not contain any data, but there is data in the emptydir volume? does it delete the data from the emptydir volume to match the pvc? or does it only copy missing data?

Reply:
i'll have to check and see.  the old minio would be running with a fresh and empty data dir in that case, so ideally it would not remove anything from the destination.  my guess would be it does not be default given there is a `--remove` flag:
```
NAME:                                                                                                            
  mc mirror - synchronize object(s) to a remote site                                                             
                                                                                                                 
USAGE:                                                                                                           
  mc mirror [FLAGS] SOURCE TARGET                                                                                
                                                                                                                 
FLAGS:                                                                                                           
...                             
  --remove                           remove extraneous object(s) on target 
```

Reply:
nice, i hope that's the case

Reply:
old minio would not have a bucket, so the export couldn't even occur.  this will be a no-op